python/dolfinx/fem/function.py

```diff
@@ -260,7 +260,7 @@ class Function(ufl.Coefficient):
                               _cpp.fem.Function_float32, _cpp.fem.Function_float64]
 
     def __init__(self, V: FunctionSpace, x: typing.Optional[la.Vector] = None,
-                 name: typing.Optional[str] = None, dtype: npt.DTypeLike = None):
+                 name: typing.Optional[str] = None, dtype: typing.Optional[npt.DTypeLike] = None):
```

Comment:
`npt.DTypeLike` apparently already includes a `typing.Union[...., None]`, so technically this may not be required. If we want to make use of this fact, I will revert this change, but for consistency I will also remove `typing.Optional` from the other occurrence of `dtype = None` in this file.